### DISCUSSION TITLE
Problema certificati HA

### DISCUSSION THREAD

**Discussion by nikpait** [indomus/forum#104] [HUB: Home Assistant]
Ciao a tutti, ho un problema con il rinnovo dei certificati.
Ogni volta che i certificati si rinnovano, il mio HA non funziona +. Ho scoperto qual'è il problema e cioè che i certificati vengono creati come root  e non con l'utente homeassistant. Fino all'ultimo rinnovo facevo a mano la procedura di dare i diritti all'utente homeassistant (mi piacerebbe capire come farlo fare in automatico) e tutto funzionava nuovamente dopo un riavvio. Stamattina ho rifatto la procedura ma non funziona +. Nel log di HA ho qesto messaggio di errore:

ERROR (MainThread) [homeassistant.components.http] Could not read SSL certificate from /home/homeassistant/dehydrated/certs/xxxxxxx.duckdns.org/fullchain.pem: [SSL] PEM lib (_ssl.c:4044)

I permessi mi sembrano a posto, cosa può essere?

-rwxrwxrwx 1 homeassistant homeassistant 1587 feb  4 01:00 chain-1612396816.pem
-rwxrwxrwx 1 homeassistant homeassistant 1667 apr  6 01:00 cert-1617663603.csr
-rwxrwxrwx 1 homeassistant homeassistant 3791 apr  6 01:00 fullchain-1617663603.pem
-rwxrwxrwx 1 homeassistant homeassistant 1587 apr  6 01:00 chain-1617663603.pem
-rwxrwxrwx 1 homeassistant homeassistant 2204 apr  6 01:00 cert-1617663603.pem
lrwxrwxrwx 1 pi            pi              22                      apr  6 01:00 privkey.pem -> privkey-1617663603.pem
lrwxrwxrwx 1 pi            pi              24                      apr  6 01:00 fullchain.pem -> fullchain-1617663603.pem
lrwxrwxrwx 1 pi            pi              20                      apr  6 01:00 chain.pem -> chain-1617663603.pem
lrwxrwxrwx 1 pi            pi              19                      apr  6 01:00 cert.pem -> cert-1617663603.pem
lrwxrwxrwx 1 pi            pi              19                      apr  6 01:00 cert.csr -> cert-1617663603.csr
-rwxrwxrwx 1 homeassistant homeassistant    0    apr  6 08:27 privkey-1617663603.pem

Grazie

**Comment by lorenzopiazzi** [comment]
Ciao @nikpait , anch'io ho avuto un problema analogo, i certificati si erano rinnovati ma HA non "lo sapeva"... ho inserito nel config:
http:
  ssl_certificate: /ssh/fullchain.pem
  ssl_key: privkey.pem
al che HA si è reso conto che i certificati si erano aggiornati.
Sarò onesto: questo mi ha creato un po' di problemi di autenticazione, evidentemente non ho ancora una configurazione robusta di let's encrypt ma ho voluto ugualmente risponderti nel caso riterrai di fare un tentativo.

**Comment by nikpait** [comment]
Ciao, ho risolto il problema della lettura del file pem, facendo ricreare i certificati e ridando i permessi all'utente homeassistant.
Rimane il problema che i certificati vengono creati con l'utente root e quindi non leggibili dal sistema.
